src/kmerseek/search.py

```diff
@@ -0,0 +1,286 @@
+import click
+from sourmash_plugin_branchwater import sourmash_plugin_branchwater
+import pandas as pd
+import polars as pl
+
+from .uniprot import get_domains
+from .sketch import make_sketch_kws
+from .index import KmerseekIndex, _make_siglist_file
+from .query import KmerseekQuery
+from .logging import setup_logging, logger
+
+
+def show_results(results_per_gene):
+    for gene, match in results_per_gene:
+        domains = get_domains(gene, match.target_start, match.target_end)
+
+        logger.info(
+            f"Found {match.query_name}:{match.query_start}-{match.query_end} "
+            f"in {match.target_name}:{match.target_start}-{match.target_end}"
+        )
+        if domains:
+            for domain in domains:
+                logger.info(f"Found: {domain.name} in {domain.start}-{domain.end}")
+        logger.info(match.query_seq)
+        logger.info(match.hp_kmer)
+        logger.info(match.target_seq)
+
+
+def single_stitch_together_kmers(kmers: pl.Series, i_kmers: pl.Series):
+    stitched = ""
+    logger.debug(f"Input kmers: {kmers}")
+    logger.debug(f"Input i_kmers: {i_kmers}")
+
+    for i, (i_kmer, kmer) in enumerate(zip(i_kmers, kmers)):
+        logger.debug(f"Processing kmer {i}: position={i_kmer}, kmer={kmer}")
+        if i == 0:
+            stitched = kmer
+            prev_i_kmer = i_kmer
+            logger.debug(f"First kmer: {stitched}")
+        else:
+            kmer_slice = i_kmer - prev_i_kmer
+            logger.debug(f"Slice size: {kmer_slice}")
+            stitched += kmer[-kmer_slice:]
+            logger.debug(f"After adding slice: {stitched}")
+        prev_i_kmer = i_kmer
+
+    logger.debug(f"Final stitched sequence: {stitched}")
+    return stitched
+
+
+def stitch_kmers_in_query_match_pair(
+    df: pl.LazyFrame,
+    kmer_match="kmer_match",
+    kmer_alphabet="encoded",
+    kmer_query="kmer_query",
+    start_match="start_match",
+    start_query="start_query",
+) -> pl.DataFrame:
+    logger.debug("\nProcessing new group:")
+    logger.debug(f"Input dataframe:\n{df}")
+
+    df = df.sort(start_query, descending=False)
+    logger.debug(f"Sorted dataframe:\n{df}")
+
+    match_name = df["match_name"][0]
+    query_name = df["query_name"][0]
+
+    query = single_stitch_together_kmers(df[kmer_query], df[start_match])
+    alphabet = single_stitch_together_kmers(df[kmer_alphabet], df[start_query])
+    match = single_stitch_together_kmers(df[kmer_match], df[start_match])
+
+    logger.debug(f"query: {query}")
+    logger.debug(f"alpha: {alphabet}")
+    logger.debug(f"match: {match}")
+
+    # All these lengths should be the same -> if not, something went wrong
+    assert len(query) == len(alphabet)
+    assert len(alphabet) == len(match)
+
+    length = len(query)
+
+    match_start = df[start_match].min()
+    query_start = df[start_query].min()
+    match_end = match_start + length
```

Comment:
Would be nice to have the coordinate system explicitly documented somewhere for these, I assume 1-based fully closed intervals?